packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -236,6 +243,93 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
   return self;
 }
 
+- (void)setPictureInPicture:(BOOL)pictureInPicture {
+    if (self._pictureInPicture == pictureInPicture) {
+        return;
+    }
+
+    self._pictureInPicture = pictureInPicture;
+    if (@available(iOS 9.0, *)) {
+        if (_pipController && self._pictureInPicture && ![_pipController isPictureInPictureActive]) {
+            self._playerLayer.opacity = 0.001;
+            if (_eventSink != nil) {
+              _eventSink(@{@"event" : @"startingPiP"});
+            }
+            dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
will `setPictureInPicture` ever be called on a background thread?

Reply:
Not 100% sure. But it will be called over a method channel. I always try to dispatch to the main thread when using method channels. Isn't that something I should do?

Reply:
You can verify by setting a breakpoint there and see which thread it is on. IIRC method channel calls platform code on the main thread, in which case you shouldn't call call this `dispatch_async`

Reply:
> IIRC method channel calls platform code on the main thread

Correct; unless you explicitly opt a channel into the new background thread system it will always be called on the main thread.

Reply:
Removed the dispatch_async